src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/CompletionSet2.cs

```diff
@@ -101,13 +108,14 @@ internal void SetTrackingSpan(ITrackingSpan trackingSpan)
         private VSCompletion ConvertCompletionItem(CompletionItem item, string displayTextOpt = null)
         {
             VSCompletion value;
-            if (!_completionItemMap.TryGetValue(item, out value))
+            if (!_completionItemToVSCompletion.TryGetValue(item, out value))
             {
                 value = new CustomCommitCompletion(
                     _completionPresenterSession,
                     item,
                     displayTextOpt ?? item.DisplayText);
-                _completionItemMap.Add(item, value);
+                _completionItemToVSCompletion.Add(item, value);
+                value.Properties.AddProperty(typeof(CompletionItem), item);
```

Comment:
I don't even buy the comment below.  How can one dictionary not be an issue.  But if we had two dictionaries, it would *now* suddenly be an issue.  IMO just having two dictionaries would be fine.  I just used the property bag since it was nice and available.